src/main/java/github/scarsz/discordsrv/DiscordSRV.java

```diff
@@ -1790,16 +1791,32 @@ public void processChatMessage(Player player, Component message, String channel,
         String username = player.getName();
         if (!reserializer) username = DiscordUtil.escapeMarkdown(username);
 
-        String discordMessage = (hasGoodGroup
-                ? LangUtil.Message.CHAT_TO_DISCORD.toString()
-                : LangUtil.Message.CHAT_TO_DISCORD_NO_PRIMARY_GROUP.toString())
-                .replaceAll("%time%|%date%", TimeUtil.timeStamp())
-                .replace("%channelname%", channel != null ? channel.substring(0, 1).toUpperCase() + channel.substring(1) : "")
-                .replace("%primarygroup%", userPrimaryGroup)
-                .replace("%username%", username)
-                .replace("%usernamenoescapes%", MessageUtil.strip(player.getName()))
-                .replace("%world%", player.getWorld().getName())
-                .replace("%worldalias%", MessageUtil.strip(MultiverseCoreHook.getWorldAlias(player.getWorld().getName())));
+        String discordMessage = "";
+
+        // Use a different format if the message is a LunaChat converted message
+        if (PluginUtil.pluginHookIsEnabled("LunaChat") && event.getClass().equals(LunaChatBukkitPostJapanizeEvent.class)) {
+            discordMessage = (hasGoodGroup
+                    ? LangUtil.Message.LUNACHAT_CONVERTED_TO_DISCORD.toString()
+                    : LangUtil.Message.LUNACHAT_CONVERTED_TO_DISCORD_NO_PRIMARY_GROUP.toString())
+                    .replaceAll("%time%|%date%", TimeUtil.timeStamp())
+                    .replace("%channelname%", channel != null ? channel.substring(0, 1).toUpperCase() + channel.substring(1) : "")
+                    .replace("%primarygroup%", userPrimaryGroup)
+                    .replace("%username%", username)
+                    .replace("%usernamenoescapes%", MessageUtil.strip(player.getName()))
+                    .replace("%world%", player.getWorld().getName())
+                    .replace("%worldalias%", MessageUtil.strip(MultiverseCoreHook.getWorldAlias(player.getWorld().getName())));
+        } else {
+            discordMessage = (hasGoodGroup
+                    ? LangUtil.Message.CHAT_TO_DISCORD.toString()
+                    : LangUtil.Message.CHAT_TO_DISCORD_NO_PRIMARY_GROUP.toString())
+                    .replaceAll("%time%|%date%", TimeUtil.timeStamp())
+                    .replace("%channelname%", channel != null ? channel.substring(0, 1).toUpperCase() + channel.substring(1) : "")
+                    .replace("%primarygroup%", userPrimaryGroup)
+                    .replace("%username%", username)
+                    .replace("%usernamenoescapes%", MessageUtil.strip(player.getName()))
+                    .replace("%world%", player.getWorld().getName())
+                    .replace("%worldalias%", MessageUtil.strip(MultiverseCoreHook.getWorldAlias(player.getWorld().getName())));
+        }
```

Comment:
Would like to avoid referencing the LunaChat class here as it may cause issues if LunaChat isn't being used. (Probably by just passing the message formats in as parameters to the `processChatMessage` method)

The code in the if and else blocks are nearly identical, we should avoid copying the code, by moving the replace calls outside of the if, example:
```java
if (... lunachat check...) {
  discordMessage = ...lunachat format...;
} else {
  discordMessage = ...normal format...;
}

discordMessage = discordMessage
  .replace("...", ...)
  .replace("...", ...)
...
```